### PR TITLE
Added Java implementation

### PR DESCRIPTION
**Pull request by Seregy**

### CODE REVIEW

File: Java/SimpleEventEmitter.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * @param <T> type of the listener's parameter
+ */
+public class SimpleEventEmitter<T> {
+    /**
+     * Map with all registered events and corresponding listeners.
+     */
+    private final Map<String, List<Consumer<T>>> events;
+
+    /**
+     * Constructs new SimpleEventEmitter.
```
Comment (lundibundi) [inline Java/SimpleEventEmitter.java]: This comment seems redundant to me.

File: Java/SimpleEventEmitter.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * Fires event with given data, executing all event listeners.
+     *
+     * @param name event's name
+     * @param data event's data
+     */
+    public void emit(final String name, final T data) {
+        List<Consumer<T>> listeners = events.get(name);
+        if (listeners != null) {
+            for (Consumer<T> listener : listeners) {
```
Comment (lundibundi) [inline Java/SimpleEventEmitter.java]: As you use java 8 in this class, I'd suggest you use stream api for this one.

File: Java/SimpleEventEmitter.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        return actions != null;
+    }
+
+
+    /**
+     * Entry point of the program, used for testing.
+     *
+     * @param args command-line arguments, not used
+     */
+    public static void main(final String[] args) {
```
Comment (lundibundi) [inline Java/SimpleEventEmitter.java]: I'd like test methods to be in separate class from this lib-class, probably written with junit.

**Review by lundibundi — CHANGES_REQUESTED** [review]
Overall looks good with a few changes. 

Also I'd suggest to try `Optional<>` with `events` this may simplify other methods but that's not required and I'll LGTM even without it.

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
Rename files to `N-name.java` like in JS

**Comment by Seregy** [comment]
I've renamed files and made almost all proposed changes. I didn't write test methods because I found it impossible to do with junit inside non-public class(and the class is non-public because of non-conventional file name). So I just moved usage example into separate file. Should I write tests in separate file or pick another testing framework?

**Comment by lundibundi** [comment]
@tshemsedinov I'd suggest to rename library class back to something like SimpleEventEmitter as it is done in C# version and the usage/test file name may be in JS style. Because right now it is extremely strange and holds no meaning. Or we can make the library class 'executable' as it was before and name the file 1-simple.

EDIT: @Seregy let's wait for @tshemsedinov to respond to this one and decide based on that.

**Review by lundibundi — APPROVED** [review]
LGTM after resolution of filenames problem. Thanks.

**Review by lidaamber — APPROVED** [review]

**Pull request merged**